### DISCUSSION TITLE
How to autoload a non-class .php file

### DISCUSSION THREAD

**Discussion by ahmu83** [composer/composer#11260] [Q&A / Support] [answered]
I have added psr-4 autoloading in my `composer.json` like this, and I would like to know how to autoload a non-class .php file. i.e a file that contains just functions.

```
    "autoload": {
        "psr-4": {
            "App\\": "inc/App"
        }
    }

```

Everything is working fine but I want to know is there a way to autoload a non-class .php file with namespacing?

i.e, my `test-functions.php` has a namespace of `namespace App\TestFunctions` 
and I would like this to be included when I do `use App\TestFunctions`

Is this possible?

**Comment by stof** [comment, marked as answer]
PHP does not have an autoloader for functions. They must be loaded before usage.

However, it is possible to tell composer to load some files at the time `vendor/autoload.php` is included, which is useful for files defining functions (but they will be loaded early, not only when they are used). For that, use the `files` key under `autoload` in your config: https://getcomposer.org/doc/04-schema.md#files
